src/blockdata/script.rs

```diff
@@ -637,6 +637,26 @@ pub struct Instructions<'a> {
 }
 
 impl<'a> Instructions<'a> {
+    /// Set the iterator to end so that it won't iterate any longer
+    fn kill(&mut self) {
+        let len = self.data.len();
+        self.data.nth(len.max(1) - 1);
+    }
+
+    /// takes `len` bytes long slice from iterator and returns it advancing iterator
+    /// if the iterator is not long enough `None` is returned and the iterator is killed
+    /// to avoid returning an infinite stream of errors.
+    fn take_slice_or_kill(&mut self, len: usize) -> Option<&'a [u8]> {
```

Comment:
In f376391d51ac68d21fc45fc4f2b7034b7aecf64c:

I like this, but I think it should return a `Result` and directly return an `Err(EarlyEndOfScript)` so you don't need `map_err` on every call site.